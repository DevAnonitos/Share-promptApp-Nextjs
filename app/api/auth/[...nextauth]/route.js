import NextAuth from "next-auth/next";
import GoogleProvider from "next-auth/providers/google";

const handler = NextAuth({
    providers: [
        GoogleProvider({
            clientId: process.env.GOOGLE_ID,
            clientSecret: process.env.GOOGLE_CLIENT_SECRET,
        })
    ],
    callbacks: {
        async session({ session }) {

        },
        async signIn({ account, profile, user, credentials }) {
            try {

            } catch (error) {
                console.log("Error checking if user exists: ", error.message);
                return false;
            }
        },
    }
});

export {
    handler as GET,
    handler as POST,
};
