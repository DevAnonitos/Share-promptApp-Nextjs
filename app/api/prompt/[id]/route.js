import Prompt from "../../../../models/prompt";
import { connectToDB } from "../../../../utils/database";

export const GET = async (request, { params }) => {
    try {
        await connectToDB();

        const prompt = await Prompt.findById(params.id).populate("creator");

        if(!prompt) {
            return new Response("Prompt not found", {
                status: 404,
            });
        }

        return new Response(JSON.stringify(prompt), {
            status: 200,
        });
    } catch (error) {
        return new Response("Internal Server Error", {
            status: 500,
        });
    }
};

export const PATH = async (request, { params }) => {
    const { prompt, tag } = await request.json();

    try {
        await connectToDB();

        const existingPrompt = await Prompt.findById(params.id);

        if(!existingPrompt) {
            return new Response("Prompt not founded", {
                status: 404,
            });
        }

        existingPrompt.prompt = prompt;
        existingPrompt.tag = tag;

        return new Response("Successfully update-prompt!", {
            status: 200,
        });

    } catch (error) {
        return new Response("Error updating Prompt", {
            status: 500,
        });
    }
};

export const DELETE = async (request, { params }) => {
    try {
        await connectToDB();

        await Prompt.findByIdAndRemove(params.id);

        return new Response("Prompt delete successfully", {
            status: 200,
        });
    } catch (error) {
        return new Response("Error deleting prompt", {
            status: 500,
        });
    }
};
