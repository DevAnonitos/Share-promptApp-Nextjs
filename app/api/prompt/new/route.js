import { connectToDB } from "@/utils/database";

export const POST = async (request) => {
    const { userId, prompt, tag } = await request.json();

    try {
        await connectToDB();

        return new Response(JSON.stringify(), {
            status: 201,
        })
    } catch (error) {
        return new Response("Fail to create a new prompt", {
            status: 500,
        })
    }
};
