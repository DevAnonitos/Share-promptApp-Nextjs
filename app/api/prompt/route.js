import Prompt from "@/models/prompt";
import { connectToDB } from "@/libs/utils/database";

export const GET = async (request) => {
    try {
        await connectToDB();

        const prompts = await Prompt.find({}).populate('creator');

        return new Response(JSON.stringify(prompts), {
            status: 200,
        });
    } catch (error) {
        return new Response(JSON("Fail to fetch All prompts", {
            status: 500,
        }))
    }
};
