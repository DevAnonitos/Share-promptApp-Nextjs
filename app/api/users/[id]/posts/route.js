import Prompt from "@/libs/models/prompt";
import { connectToDB } from "@/libs/utils/database";

export const GET = async (request, { params }) => {
    try {
        await connectToDB();

        if(!params || !params.id) {
            return new Response("UserId is missing", {
                status: 400,
            });
        }

        const prompts = await Prompt.find({ creator: params.id }).populate("creator");

        return new Response(JSON.stringify(prompts), {
            status: 200,
        });
    } catch (error) {
        console.log(error);
        return new Response("Failed to fetch prompts created by user", {
            status: 500
        });
    }
};
