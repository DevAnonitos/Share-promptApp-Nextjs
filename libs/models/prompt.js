import { Schema, model, models } from "mongoose";

// Create PromptSchema Mongoose
const PromptSchema = new Schema({
    creator: {
        type: Schema.Types.ObjectId,
        ref: 'User',
    },
    prompt: {
        type: String,
        require: [
            true,
            'Prompt is required!',
        ],
    },
    tag: {
        type: String,
        required: [
            true,
            'Tag is required!',
        ],
    },
});

const Prompt  = models.Prompt || model('Prompt', PromptSchema);

export default Prompt;
