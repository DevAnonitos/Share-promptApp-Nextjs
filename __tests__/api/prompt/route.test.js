/**
* @jest-environment node
*/
import { GET } from "@/app/api/prompt/route";

// jest.mock('@/libs/models/prompt'); 

// export const mockDatabase = {
//   connect: jest.fn().mockResolvedValueOnce(), // Mock successful connection
//   find: jest.fn(), // Mock find method for queries
// };

// export const connectToDB = async () => {
//   await mockDatabase.connect(); // Use mocked connection
// };

describe('API Prompt', () => {
  // Test fetch fail response all prompt
  it('should [GET]: a fail response data as JSON from API Route', async () => {
    // mockDatabase.connect.mockResolvedValueOnce();

    // // Mock Prompt.find to return expected data (replace with your structure)
    // const mockFind = jest.fn().mockResolvedValueOnce([{ id: 1, name: "Prompt 1" }]);
    // Prompt.find = mockFind; 

    // const res = await GET();
    // const data = await res.json();

    // expect(res.status).toEqual(200); // Expect a successful response (200)
    // expect(data).toBeDefined();
    // expect(data.length).toBe(1); //
    
  });

  it("should [GET]: a fail response data as JSON from API Route", async () => {
    const res = await GET();
    const data = await res.json();
    console.log(data);
    expect(res.status).toEqual(500); 
    expect(data).toBeDefined();
  })

});
