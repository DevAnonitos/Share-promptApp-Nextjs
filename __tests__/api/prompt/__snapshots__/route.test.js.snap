// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`API Prompt should [GET]: a fail response data as JSON from API Route 1`] = `"Fail to fetch All prompts"`;

exports[`API Prompt should Response has the required fields 1`] = `true`;

exports[`API Prompt should Response has the required fields 2`] = `"text/plain;charset=UTF-8"`;

exports[`API Prompt should Response status code is 200 1`] = `200`;

exports[`API Prompt should return data json from API Prompt 1`] = `
[
  {
    "__v": 0,
    "_id": "645cee833a66a9ea8f5f46bd",
    "creator": {
      "__v": 0,
      "_id": "645a5ff09c6a32502c433959",
      "email": "nguyebao281003@gmail.com",
      "image": "https://lh3.googleusercontent.com/a/AGNmyxZ3gxWO5zNF-XeG---qMq4521lDBjTF0ZUtxJk98w=s96-c",
      "username": "nguyebao281003",
    },
    "prompt": "test",
    "tag": "         ",
  },
  {
    "__v": 0,
    "_id": "645ceef83a66a9ea8f5f46c3",
    "creator": {
      "__v": 0,
      "_id": "645a5ff09c6a32502c433959",
      "email": "nguyebao281003@gmail.com",
      "image": "https://lh3.googleusercontent.com/a/AGNmyxZ3gxWO5zNF-XeG---qMq4521lDBjTF0ZUtxJk98w=s96-c",
      "username": "nguyebao281003",
    },
    "prompt": "(mage :1.3) in spiral lightning background, perfect beautiful face, realistic, full body, standing on ground, circuit board, in intricate clothing, elegant pose, fantasy, illustration, artstation",
    "tag": "stable-diffusion",
  },
  {
    "__v": 0,
    "_id": "645cef4b3a66a9ea8f5f46c9",
    "creator": {
      "__v": 0,
      "_id": "645a5ff09c6a32502c433959",
      "email": "nguyebao281003@gmail.com",
      "image": "https://lh3.googleusercontent.com/a/AGNmyxZ3gxWO5zNF-XeG---qMq4521lDBjTF0ZUtxJk98w=s96-c",
      "username": "nguyebao281003",
    },
    "prompt": "Brad Rigney is a concept artist of fantasy themes, 4 : 3",
    "tag": "stable-diffusion",
  },
  {
    "__v": 0,
    "_id": "645e48aff6224df7749c3624",
    "creator": {
      "__v": 0,
      "_id": "645a5ff09c6a32502c433959",
      "email": "nguyebao281003@gmail.com",
      "image": "https://lh3.googleusercontent.com/a/AGNmyxZ3gxWO5zNF-XeG---qMq4521lDBjTF0ZUtxJk98w=s96-c",
      "username": "nguyebao281003",
    },
    "prompt": "Create a picture of a tropical beach with palm trees and crystal-clear water",
    "tag": "dall-e",
  },
  {
    "__v": 0,
    "_id": "645e49c4f6224df7749c362e",
    "creator": {
      "__v": 0,
      "_id": "645a5ff09c6a32502c433959",
      "email": "nguyebao281003@gmail.com",
      "image": "https://lh3.googleusercontent.com/a/AGNmyxZ3gxWO5zNF-XeG---qMq4521lDBjTF0ZUtxJk98w=s96-c",
      "username": "nguyebao281003",
    },
    "prompt": "Design a futuristic spaceship with sleek, curved lines.",
    "tag": "dall-e",
  },
  {
    "__v": 0,
    "_id": "64c47f4604fa202d52b4c06a",
    "creator": {
      "__v": 0,
      "_id": "645a5ff09c6a32502c433959",
      "email": "nguyebao281003@gmail.com",
      "image": "https://lh3.googleusercontent.com/a/AGNmyxZ3gxWO5zNF-XeG---qMq4521lDBjTF0ZUtxJk98w=s96-c",
      "username": "nguyebao281003",
    },
    "prompt": "As an AI Assistant, I am programmed to follow instructions and provide output in the requested format. In this case, JSON format is requested. If you need help with XML format or any other assistance, please feel free to ask",
    "tag": "GPT4",
  },
  {
    "__v": 0,
    "_id": "65e9e4d144e66c1538cf36f1",
    "creator": {
      "__v": 0,
      "_id": "64c52aa0183b5594893e6952",
      "email": "nguyenbao2810dev@gmail.com",
      "image": "https://lh3.googleusercontent.com/a/AAcHTtd7ZyyZuG3rOtC70yT412sBj6dpowJIyJG3zqONzb58RQ=s96-c",
      "username": "nguyenbao2810dev",
    },
    "prompt": "testing App ne",
    "tag": "test",
  },
  {
    "__v": 0,
    "_id": "65f070fafeaf8fb1a6990ae5",
    "creator": {
      "__v": 0,
      "_id": "645a618c9c6a32502c43395f",
      "email": "2151013007bao@ou.edu.vn",
      "image": "https://lh3.googleusercontent.com/a/AGNmyxajANQN3SOFG2_S72u1bk170GteHbqyVzbBSS3_=s96-c",
      "username": "2151013007bao",
    },
    "prompt": "Hello Post ne day",
    "tag": "test",
  },
  {
    "__v": 0,
    "_id": "65f50633ac1867d4610726cc",
    "creator": {
      "__v": 0,
      "_id": "64c52aa0183b5594893e6952",
      "email": "nguyenbao2810dev@gmail.com",
      "image": "https://lh3.googleusercontent.com/a/AAcHTtd7ZyyZuG3rOtC70yT412sBj6dpowJIyJG3zqONzb58RQ=s96-c",
      "username": "nguyenbao2810dev",
    },
    "prompt": "HDR photo of woman with hair in a messy bun smiles at the viewer, perfect eyes, beautiful skin, outside on a warm sunny day wearing a summer dress . High dynamic range, vivid, rich details, clear shadows and highlights, realistic, intense, enhanced contrast, highly detailed",
    "tag": "stable diffusion",
  },
  {
    "__v": 0,
    "_id": "65f506c8ac1867d4610726d4",
    "creator": {
      "__v": 0,
      "_id": "64c52aa0183b5594893e6952",
      "email": "nguyenbao2810dev@gmail.com",
      "image": "https://lh3.googleusercontent.com/a/AAcHTtd7ZyyZuG3rOtC70yT412sBj6dpowJIyJG3zqONzb58RQ=s96-c",
      "username": "nguyenbao2810dev",
    },
    "prompt": "A portrait of a young woman with sun-kissed blonde wavy hair and light hazel eyes. She has a clear complexion with a scattering of freckles across her nose and cheeks. Her expression is contemplative as she gazes towards the viewer. She wears minimal makeup, with a focus on her eyes, and an elegant white or silver drop earring. She is in a simple black sleeveless top, against a soft, natural background suggesting an outdoor setting. The lighting is soft and natural, highlighting her features with a serene ambiance.
  
",
    "tag": "stable diffusion",
  },
  {
    "__v": 0,
    "_id": "65f8533841345b0a2dbbd386",
    "creator": {
      "__v": 0,
      "_id": "645a5ff09c6a32502c433959",
      "email": "nguyebao281003@gmail.com",
      "image": "https://lh3.googleusercontent.com/a/AGNmyxZ3gxWO5zNF-XeG---qMq4521lDBjTF0ZUtxJk98w=s96-c",
      "username": "nguyebao281003",
    },
    "prompt": "Hello tets",
    "tag": "ai",
  },
  {
    "__v": 0,
    "_id": "65f9573bdeb672384b1688b5",
    "creator": {
      "__v": 0,
      "_id": "645a5ff09c6a32502c433959",
      "email": "nguyebao281003@gmail.com",
      "image": "https://lh3.googleusercontent.com/a/AGNmyxZ3gxWO5zNF-XeG---qMq4521lDBjTF0ZUtxJk98w=s96-c",
      "username": "nguyebao281003",
    },
    "prompt": "Translate regular text into emoji text.",
    "tag": "gpt",
  },
  {
    "__v": 0,
    "_id": "65f9574fdeb672384b1688ba",
    "creator": {
      "__v": 0,
      "_id": "645a5ff09c6a32502c433959",
      "email": "nguyebao281003@gmail.com",
      "image": "https://lh3.googleusercontent.com/a/AGNmyxZ3gxWO5zNF-XeG---qMq4521lDBjTF0ZUtxJk98w=s96-c",
      "username": "nguyebao281003",
    },
    "prompt": "Create tables from unstructured text.",
    "tag": "gpt",
  },
  {
    "__v": 0,
    "_id": "6638280afd1d37dbc2c8ae79",
    "creator": {
      "__v": 0,
      "_id": "6625d3939ec5778aaf9eb914",
      "email": "lythanhkhoa360@gmail.com",
      "image": "https://lh3.googleusercontent.com/a/ACg8ocL6h9HDuVg2qtgc7-cjVfNcJZ3xQt_3hGdDuK-Y-U6NSzvGrQ=s96-c",
      "username": "lythanhkhoa360",
    },
    "prompt": "",
    "tag": "abcasfsdaf",
  },
  {
    "__v": 0,
    "_id": "66382872fd1d37dbc2c8ae7b",
    "creator": {
      "__v": 0,
      "_id": "6625d3939ec5778aaf9eb914",
      "email": "lythanhkhoa360@gmail.com",
      "image": "https://lh3.googleusercontent.com/a/ACg8ocL6h9HDuVg2qtgc7-cjVfNcJZ3xQt_3hGdDuK-Y-U6NSzvGrQ=s96-c",
      "username": "lythanhkhoa360",
    },
    "prompt": "1234214",
    "tag": "abcasfsdaf",
  },
  {
    "__v": 0,
    "_id": "6638289cfd1d37dbc2c8ae7d",
    "creator": {
      "__v": 0,
      "_id": "6625d3939ec5778aaf9eb914",
      "email": "lythanhkhoa360@gmail.com",
      "image": "https://lh3.googleusercontent.com/a/ACg8ocL6h9HDuVg2qtgc7-cjVfNcJZ3xQt_3hGdDuK-Y-U6NSzvGrQ=s96-c",
      "username": "lythanhkhoa360",
    },
    "prompt": "1234214",
    "tag": "abcasfsdaf",
  },
  {
    "__v": 0,
    "_id": "663828fcfd1d37dbc2c8ae7f",
    "creator": {
      "__v": 0,
      "_id": "6625d3939ec5778aaf9eb914",
      "email": "lythanhkhoa360@gmail.com",
      "image": "https://lh3.googleusercontent.com/a/ACg8ocL6h9HDuVg2qtgc7-cjVfNcJZ3xQt_3hGdDuK-Y-U6NSzvGrQ=s96-c",
      "username": "lythanhkhoa360",
    },
    "prompt": "1234214",
    "tag": "abcasfsdaf",
  },
  {
    "__v": 0,
    "_id": "66382909fd1d37dbc2c8ae85",
    "creator": {
      "__v": 0,
      "_id": "6625d3939ec5778aaf9eb914",
      "email": "lythanhkhoa360@gmail.com",
      "image": "https://lh3.googleusercontent.com/a/ACg8ocL6h9HDuVg2qtgc7-cjVfNcJZ3xQt_3hGdDuK-Y-U6NSzvGrQ=s96-c",
      "username": "lythanhkhoa360",
    },
    "prompt": "1234214",
    "tag": "abcasfsdaf",
  },
  {
    "__v": 0,
    "_id": "66382944fd1d37dbc2c8ae87",
    "creator": {
      "__v": 0,
      "_id": "6625d3939ec5778aaf9eb914",
      "email": "lythanhkhoa360@gmail.com",
      "image": "https://lh3.googleusercontent.com/a/ACg8ocL6h9HDuVg2qtgc7-cjVfNcJZ3xQt_3hGdDuK-Y-U6NSzvGrQ=s96-c",
      "username": "lythanhkhoa360",
    },
    "prompt": "1234214",
    "tag": "abcasfsdaf",
  },
  {
    "__v": 0,
    "_id": "66382960fd1d37dbc2c8ae89",
    "creator": {
      "__v": 0,
      "_id": "6625d3939ec5778aaf9eb914",
      "email": "lythanhkhoa360@gmail.com",
      "image": "https://lh3.googleusercontent.com/a/ACg8ocL6h9HDuVg2qtgc7-cjVfNcJZ3xQt_3hGdDuK-Y-U6NSzvGrQ=s96-c",
      "username": "lythanhkhoa360",
    },
    "prompt": "1234214",
    "tag": "abcasfsdaf",
  },
  {
    "__v": 0,
    "_id": "66382992fd1d37dbc2c8ae8b",
    "creator": {
      "__v": 0,
      "_id": "6625d3939ec5778aaf9eb914",
      "email": "lythanhkhoa360@gmail.com",
      "image": "https://lh3.googleusercontent.com/a/ACg8ocL6h9HDuVg2qtgc7-cjVfNcJZ3xQt_3hGdDuK-Y-U6NSzvGrQ=s96-c",
      "username": "lythanhkhoa360",
    },
    "prompt": "fdafdasf",
    "tag": "abcasfsdaf",
  },
  {
    "__v": 0,
    "_id": "66382c91fd1d37dbc2c8aec2",
    "creator": {
      "__v": 0,
      "_id": "6625d3939ec5778aaf9eb914",
      "email": "lythanhkhoa360@gmail.com",
      "image": "https://lh3.googleusercontent.com/a/ACg8ocL6h9HDuVg2qtgc7-cjVfNcJZ3xQt_3hGdDuK-Y-U6NSzvGrQ=s96-c",
      "username": "lythanhkhoa360",
    },
    "prompt": "",
    "tag": "fdafa",
  },
  {
    "__v": 0,
    "_id": "66382c9dfd1d37dbc2c8aec4",
    "creator": {
      "__v": 0,
      "_id": "6625d3939ec5778aaf9eb914",
      "email": "lythanhkhoa360@gmail.com",
      "image": "https://lh3.googleusercontent.com/a/ACg8ocL6h9HDuVg2qtgc7-cjVfNcJZ3xQt_3hGdDuK-Y-U6NSzvGrQ=s96-c",
      "username": "lythanhkhoa360",
    },
    "prompt": "fdafdasf",
    "tag": "fdafa",
  },
  {
    "__v": 0,
    "_id": "66382ca2fd1d37dbc2c8aec6",
    "creator": {
      "__v": 0,
      "_id": "6625d3939ec5778aaf9eb914",
      "email": "lythanhkhoa360@gmail.com",
      "image": "https://lh3.googleusercontent.com/a/ACg8ocL6h9HDuVg2qtgc7-cjVfNcJZ3xQt_3hGdDuK-Y-U6NSzvGrQ=s96-c",
      "username": "lythanhkhoa360",
    },
    "prompt": "",
    "tag": "fdafa",
  },
  {
    "__v": 0,
    "_id": "66383141fd1d37dbc2c8aed6",
    "creator": {
      "__v": 0,
      "_id": "6625d3939ec5778aaf9eb914",
      "email": "lythanhkhoa360@gmail.com",
      "image": "https://lh3.googleusercontent.com/a/ACg8ocL6h9HDuVg2qtgc7-cjVfNcJZ3xQt_3hGdDuK-Y-U6NSzvGrQ=s96-c",
      "username": "lythanhkhoa360",
    },
    "prompt": "",
    "tag": "fdafa",
  },
  {
    "__v": 0,
    "_id": "66383201fd1d37dbc2c8aed8",
    "creator": {
      "__v": 0,
      "_id": "6625d3939ec5778aaf9eb914",
      "email": "lythanhkhoa360@gmail.com",
      "image": "https://lh3.googleusercontent.com/a/ACg8ocL6h9HDuVg2qtgc7-cjVfNcJZ3xQt_3hGdDuK-Y-U6NSzvGrQ=s96-c",
      "username": "lythanhkhoa360",
    },
    "prompt": "",
    "tag": "fdafa",
  },
  {
    "__v": 0,
    "_id": "6638320cfd1d37dbc2c8aeda",
    "creator": {
      "__v": 0,
      "_id": "6625d3939ec5778aaf9eb914",
      "email": "lythanhkhoa360@gmail.com",
      "image": "https://lh3.googleusercontent.com/a/ACg8ocL6h9HDuVg2qtgc7-cjVfNcJZ3xQt_3hGdDuK-Y-U6NSzvGrQ=s96-c",
      "username": "lythanhkhoa360",
    },
    "prompt": "",
    "tag": "fdafa",
  },
  {
    "__v": 0,
    "_id": "6638321dfd1d37dbc2c8aedc",
    "creator": {
      "__v": 0,
      "_id": "6625d3939ec5778aaf9eb914",
      "email": "lythanhkhoa360@gmail.com",
      "image": "https://lh3.googleusercontent.com/a/ACg8ocL6h9HDuVg2qtgc7-cjVfNcJZ3xQt_3hGdDuK-Y-U6NSzvGrQ=s96-c",
      "username": "lythanhkhoa360",
    },
    "prompt": "",
    "tag": "fdafa",
  },
  {
    "__v": 0,
    "_id": "663832a9fd1d37dbc2c8aee2",
    "creator": {
      "__v": 0,
      "_id": "6625d3939ec5778aaf9eb914",
      "email": "lythanhkhoa360@gmail.com",
      "image": "https://lh3.googleusercontent.com/a/ACg8ocL6h9HDuVg2qtgc7-cjVfNcJZ3xQt_3hGdDuK-Y-U6NSzvGrQ=s96-c",
      "username": "lythanhkhoa360",
    },
    "prompt": "",
    "tag": "fdsafdsa",
  },
  {
    "__v": 0,
    "_id": "663834b5fd1d37dbc2c8af2d",
    "creator": {
      "__v": 0,
      "_id": "6625d3939ec5778aaf9eb914",
      "email": "lythanhkhoa360@gmail.com",
      "image": "https://lh3.googleusercontent.com/a/ACg8ocL6h9HDuVg2qtgc7-cjVfNcJZ3xQt_3hGdDuK-Y-U6NSzvGrQ=s96-c",
      "username": "lythanhkhoa360",
    },
    "prompt": "",
    "tag": "fdsafdsa",
  },
  {
    "__v": 0,
    "_id": "66384a49fd1d37dbc2c8b165",
    "creator": {
      "__v": 0,
      "_id": "6625d3939ec5778aaf9eb914",
      "email": "lythanhkhoa360@gmail.com",
      "image": "https://lh3.googleusercontent.com/a/ACg8ocL6h9HDuVg2qtgc7-cjVfNcJZ3xQt_3hGdDuK-Y-U6NSzvGrQ=s96-c",
      "username": "lythanhkhoa360",
    },
    "prompt": "{"$gt": undefined}",
    "tag": "{"$gt": undefined}",
  },
  {
    "__v": 0,
    "_id": "6641d76fe30b79981d1904b0",
    "creator": {
      "__v": 0,
      "_id": "6641d6d2e30b79981d1904a6",
      "email": "nhom6quiz@gmail.com",
      "image": "https://lh3.googleusercontent.com/a/ACg8ocJ7N7b29KkTMYdwAkK-Vg5P4yFPpEqCt_1TEjs6RzRroRAghQ=s96-c",
      "username": "nhom6quiz",
    },
    "prompt": "ر أشقر مموج وعينين عسليتين فاتحتين. تتمتع ببشرة صافية مع تناثر النمش على أنفها وخدودها. تعبيرها تأملي وهي تنظر نحو المشاهد. وهي ترتدي الحد الأدنى من الماكياج، مع التركيز على عينيها، وقرط متدلي أنيق باللون الأبيض أو الفضي. إنها ترتدي قميصًا أسود بسيطًا بلا أكمام، على خلفية طبيعية ناعمة تشير إلى مكان خارجي. الإضاءة ناعمة وطبيعية، تبرز ملامحها بأجواء هادئة.
surat liaimra'at shabat dhat shaer 'ashqar mumawwj waeaynayn easaliatayn fatihatayni. tatamatae bibashrat safiat mae tanathur alnamsh ealaa 'anfiha wakhududiha. taebiruha tamuli wahi tanzur nahw almushahidi. wahi tartadi alhada al'adnaa min almakiaji, mae altarkiz ealaa eaynayha, waqart mutadaliy 'aniq biallawn al'abyad 'aw alfadi. 'iinaha tartadi qmysan 'aswad bsytan bila 'akmamu, ealaa khalfiat tabieiat naeimat tushir 'iilaa makan khariji. al'iida'at naeimat watabieiatan, tabruz malamihuha bi'ajwa' hadiatin.",
    "tag": "#Idea",
  },
  {
    "__v": 0,
    "_id": "6641dd59e30b79981d1904f5",
    "creator": {
      "__v": 0,
      "_id": "6641d6d2e30b79981d1904a6",
      "email": "nhom6quiz@gmail.com",
      "image": "https://lh3.googleusercontent.com/a/ACg8ocJ7N7b29KkTMYdwAkK-Vg5P4yFPpEqCt_1TEjs6RzRroRAghQ=s96-c",
      "username": "nhom6quiz",
    },
    "prompt": "???????",
    "tag": "??",
  },
  {
    "__v": 0,
    "_id": "6641f5d74ddfe9288a64a17e",
    "creator": {
      "__v": 0,
      "_id": "66384e6dfd1d37dbc2c8b1ee",
      "email": "tomalaoak@gmail.com",
      "image": "https://lh3.googleusercontent.com/a/ACg8ocJnRR3sSLv-J_BpU-tnTzcwudyoEnzdXE3bRcSQrn0htXbjWV6T=s96-c",
      "username": "tomalaoak",
    },
    "prompt": "anh da den",
    "tag": "#abcus",
  },
  {
    "__v": 0,
    "_id": "664252624ddfe9288a64a1ce",
    "creator": {
      "__v": 0,
      "_id": "66384e6dfd1d37dbc2c8b1ee",
      "email": "tomalaoak@gmail.com",
      "image": "https://lh3.googleusercontent.com/a/ACg8ocJnRR3sSLv-J_BpU-tnTzcwudyoEnzdXE3bRcSQrn0htXbjWV6T=s96-c",
      "username": "tomalaoak",
    },
    "prompt": "alo alo mot hai ba bon",
    "tag": "1234",
  },
  {
    "__v": 0,
    "_id": "66430ca31f797ddb45c2b5d3",
    "creator": {
      "__v": 0,
      "_id": "66384e6dfd1d37dbc2c8b1ee",
      "email": "tomalaoak@gmail.com",
      "image": "https://lh3.googleusercontent.com/a/ACg8ocJnRR3sSLv-J_BpU-tnTzcwudyoEnzdXE3bRcSQrn0htXbjWV6T=s96-c",
      "username": "tomalaoak",
    },
    "prompt": "anh da den",
    "tag": "abcasfsdaf",
  },
  {
    "__v": 0,
    "_id": "66430cbe1f797ddb45c2b5d7",
    "creator": {
      "__v": 0,
      "_id": "66384e6dfd1d37dbc2c8b1ee",
      "email": "tomalaoak@gmail.com",
      "image": "https://lh3.googleusercontent.com/a/ACg8ocJnRR3sSLv-J_BpU-tnTzcwudyoEnzdXE3bRcSQrn0htXbjWV6T=s96-c",
      "username": "tomalaoak",
    },
    "prompt": " ",
    "tag": "abcasfsdaf",
  },
  {
    "__v": 0,
    "_id": "66430d101f797ddb45c2b5d9",
    "creator": {
      "__v": 0,
      "_id": "66384e6dfd1d37dbc2c8b1ee",
      "email": "tomalaoak@gmail.com",
      "image": "https://lh3.googleusercontent.com/a/ACg8ocJnRR3sSLv-J_BpU-tnTzcwudyoEnzdXE3bRcSQrn0htXbjWV6T=s96-c",
      "username": "tomalaoak",
    },
    "prompt": " ",
    "tag": "abcasfsdaf",
  },
  {
    "__v": 0,
    "_id": "66430d841f797ddb45c2b5db",
    "creator": {
      "__v": 0,
      "_id": "66384e6dfd1d37dbc2c8b1ee",
      "email": "tomalaoak@gmail.com",
      "image": "https://lh3.googleusercontent.com/a/ACg8ocJnRR3sSLv-J_BpU-tnTzcwudyoEnzdXE3bRcSQrn0htXbjWV6T=s96-c",
      "username": "tomalaoak",
    },
    "prompt": " ",
    "tag": "abcasfsdaf",
  },
  {
    "__v": 0,
    "_id": "66431f819ddee6a7b4b82517",
    "creator": {
      "__v": 0,
      "_id": "66431ee69ddee6a7b4b82509",
      "email": "dfuefewf@gmail.com",
      "image": "https://lh3.googleusercontent.com/a/ACg8ocKpX278Sc8Tk7xBSXgEGpjxeOEP4xhCi_ATvZuvNjhcoM50Rw=s96-c",
      "username": "dfuefewf",
    },
    "prompt": "                                          ",
    "tag": "                     ",
  },
  {
    "__v": 0,
    "_id": "6645955c45b69910f584c3a0",
    "prompt": "Test selenium when sign in",
    "tag": "selenium test",
  },
  {
    "__v": 0,
    "_id": "6647830327f87a7b116f2827",
    "creator": {
      "__v": 0,
      "_id": "645a5ff09c6a32502c433959",
      "email": "nguyebao281003@gmail.com",
      "image": "https://lh3.googleusercontent.com/a/AGNmyxZ3gxWO5zNF-XeG---qMq4521lDBjTF0ZUtxJk98w=s96-c",
      "username": "nguyebao281003",
    },
    "prompt": "He",
    "tag": "test",
  },
]
`;
